pkg/ocibundle/native/bundle_linux.go

```diff
@@ -150,10 +181,43 @@ func (b *Bundle) Create(ctx context.Context, ociConfig *specs.Spec) error {
 		return err
 	}
 
-	// Extract from temp oci layout into bundle rootfs
-	if err := oci.UnpackRootfs(ctx, tmpDir, manifest, tools.RootFs(b.bundlePath).Path()); err != nil {
+	// Extract from temp oci layout into a temporary pristine rootfs dir, outside of the bundle.
+	// The rootfs must be nested inside a parent directory, so extracting it does not
+	// open the tmpdir permissions.
+	b.rootfsParentDir, err = os.MkdirTemp(b.tmpDir, "oci-tmp-rootfs")
+	if err != nil {
 		return err
 	}
+	pristineRootfs := filepath.Join(b.rootfsParentDir, "rootfs")
+
+	if err := oci.UnpackRootfs(ctx, tmpLayout, manifest, pristineRootfs); err != nil {
+		return err
+	}
+
+	bundleRootfs := tools.RootFs(b.bundlePath).Path()
+	if err := os.Mkdir(bundleRootfs, 0o755); err != nil && !os.IsExist(err) {
+		return err
+	}
+
+	sylog.Debugf("Performing bind mount of pristine rootfs %q to  bundle %q", pristineRootfs, bundleRootfs)
```

Comment:
The tiniest of nits: there's an extra space before the word "bundle" in the debug message 🤓